osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHit.cs

```diff
@@ -55,16 +55,8 @@ protected override void OnApply()
             type.BindValueChanged(_ =>
             {
                 updateActionsFromType();
-
-                // will overwrite samples, should only be called on subsequent changes
-                // after the initial application.
-                updateSamplesFromTypeChange();
-
                 RecreatePieces();
```

Comment:
I think what you pointed out must be related to this call, which is now happening `OnApply` and wasn't previously.

Reply:
sounds about right, base call is already recreating pieces once. not sure why the second call isn't cleaning up after the first properly, though...

Reply:
This looks to have always been an issue, and was due to the use of `Expire()` in a hierarchy where that doesn't mean the drawable is immediately removed and cleaned up (rewind logic related).

I've replaced with a `Remove` call.

Reply:
figured it'd be something along those lines... makes sense. i likely didn't notice originally since i didn't necessarily check editor too much with the original pooling implementation and forgot to follow up what with everything else going on.